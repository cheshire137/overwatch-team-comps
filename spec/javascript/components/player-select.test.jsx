import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import PlayerSelect from '../../../app/assets/javascripts/components/player-select.jsx'

describe('PlayerSelect', () => {
  let component = null
  const player1 = { id: 1, name: 'Sally' }
  const player2 = { id: 2, name: 'Jimmy' }
  let selectedPlayerID = player1.id
  let wasEditPlayerCalled = false
  let savedName = null
  const props = {
    inputID: 'input-dom-id',
    playerID: selectedPlayerID,
    players: [player1, player2],
    onChange: (newPlayerID, newName) => {
      selectedPlayerID = newPlayerID
      savedName = newName
    },
    label: 'Player 1 selection',
    editPlayer: () => { wasEditPlayerCalled = true },
    disabled: false
  }

  beforeEach(() => {
    component = <PlayerSelect {...props} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can change selected player', () => {
    const select = shallow(component).find('select')
    select.simulate('change', { target: { value: player2.id } })
    expect(selectedPlayerID).toBe(player2.id)
  })

  test('can show new player text field', () => {
    const rendered = shallow(component)

    expect(rendered.find(`#${props.inputID}`).length).toBe(0)

    const select = rendered.find('select')
    select.simulate('change', { target: { value: 'new' } })

    expect(rendered.find(`#${props.inputID}`).length).toBe(1)
  })

  test('can create new player', () => {
    const rendered = shallow(component)

    const select = rendered.find('select')
    select.simulate('change', { target: { value: 'new' } })

    const nameField = rendered.find(`#${props.inputID}`)
    nameField.simulate('change', { target: { value: 'Terry  ' } })

    const saveButton = rendered.find('.save-new-name-button')
    expect(saveButton.length).toBe(1)

    saveButton.simulate('click', { preventDefault: () => {} })
    expect(savedName).toBe('Terry')
  })

  test('can open modal to edit player', () => {
    const rendered = shallow(component)

    const button = rendered.find('.edit-player-button')
    expect(button.length).toBe(1)

    button.simulate('click', { currentTarget: { blur: () => {} } })

    expect(wasEditPlayerCalled).toBe(true)
  })
})
