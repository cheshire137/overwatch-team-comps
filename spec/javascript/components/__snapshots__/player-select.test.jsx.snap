exports[`PlayerSelect matches snapshot 1`] = `
<div
  className="player-select-container">
  <label
    className="label"
    htmlFor="input-dom-id">
    Player 1 selection
  </label>
  <div
    className="existing-player-container">
    <span
      className="select player-select ">
      <select
        disabled={false}
        onChange={[Function]}
        value={1}>
        
        <option
          value={1}>
          Sally
        </option>
        <option
          value={2}>
          Jimmy
        </option>
        <option
          value="new">
          Add new player
        </option>
      </select>
    </span>
    <button
      className="button-link edit-player-button"
      disabled={false}
      onClick={[Function]}
      type="button">
      <i
        aria-hidden="true"
        className="fa fa-cog" />
    </button>
  </div>
</div>
`;
